src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -118,6 +117,15 @@ public void checkout(
             @CheckForNull SCMRevisionState baseline)
             throws IOException, InterruptedException {
         gitSCM.checkout(build, launcher, workspace, listener, changelogFile, baseline);
+
+        BitbucketServerConfiguration server = getServer();
+        bitbucketClientFactoryProvider.getClient(server, CredentialUtils.getCredentials(server.getCredentialsId()))
+                .getBuildStatusClient(getLatestRevision(build))
```

Comment:
Javadoc for `GitSCM.getBuildData()`
```
Find the build log (BuildData) recorded with the last build that completed. BuildData
may not be recorded if an exception occurs in the plugin logic.
```
I am not sure this would always give you the current build revision. Instead of this can you use 
```
build.getEnvironment(listener).get(GitSCM.GIT_COMMIT)
```